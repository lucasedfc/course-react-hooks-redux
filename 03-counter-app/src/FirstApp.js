//import React, { Fragment } from 'react';
import React from 'react';
import PropTypes from 'prop-types';

const FirstApp = ({ name, subtitle }) => {

        
    return (
        <>
            <h1>{ name } </h1>
        {/* <pre>{ JSON.stringify(greeting, null, 3)}</pre> */}
            <p>{subtitle} </p>
        </>
        
    );

}

FirstApp.propTypes = {
    name: PropTypes.string.isRequired,
    age: PropTypes.number
}

FirstApp.defaultProps = {
    subtitle: 'Sample subtitle'
}

export default FirstApp;