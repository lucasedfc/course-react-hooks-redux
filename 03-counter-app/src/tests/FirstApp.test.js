import '@testing-library/jest-dom';
import React from 'react';
// import { render } from '@testing-library/react';
import FirstApp from '../FirstApp';
import { shallow } from 'enzyme';


describe('Test on FirstApp', () => {

    // test('should show a message "Hi Luke"', () => {
    //     const greeting = "Hi Luke";
    //     const {getByText} = render(<FirstApp name={greeting}/>)
    //     expect( getByText(greeting)).toBeInTheDocument();
    // })

    test('should show FirstApp correctly', () => {

        const greeting = 'Hi Luke'
        const wrapper = shallow(<FirstApp name={greeting} />)

        expect(wrapper).toMatchSnapshot();
    });


})
