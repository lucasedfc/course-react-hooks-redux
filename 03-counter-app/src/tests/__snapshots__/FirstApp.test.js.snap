// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on FirstApp should show FirstApp correctly 1`] = `
<Fragment>
  <h1>
    Hi Luke
    !! 
  </h1>
  <p>
    Sample subtitle
     
  </p>
</Fragment>
`;
